CMakeLists.txt

```diff
@@ -1,27 +1,122 @@
+    sdk_core/src/command_handler/lidar_command_handler.cpp
+    sdk_core/src/command_handler/command_impl.cpp
+    sdk_core/src/command_handler/command_impl.h
+    sdk_core/src/base/multiple_io/multiple_io_base.h
+    sdk_core/src/base/multiple_io/multiple_io_base.cpp
+    sdk_core/src/base/multiple_io/multiple_io_factory.h
+    sdk_core/src/base/multiple_io/multiple_io_epoll.h
+    sdk_core/src/base/multiple_io/multiple_io_epoll.cpp
+    sdk_core/src/base/multiple_io/multiple_io_poll.h
+    sdk_core/src/base/multiple_io/multiple_io_poll.cpp
+    sdk_core/src/base/multiple_io/multiple_io_select.h
+    sdk_core/src/base/multiple_io/multiple_io_select.cpp
+    sdk_core/src/base/multiple_io/multiple_io_kqueue.h
+    sdk_core/src/base/multiple_io/multiple_io_kqueue.cpp
+    sdk_core/src/base/wake_up/wake_up_pipe.h    
+)
+if(WIN32)
+	list(APPEND LIVOX_SOURCES sdk_core/src/base/network/win/network_util.cpp)
+	list(APPEND LIVOX_SOURCES sdk_core/src/base/wake_up/win/wake_up_pipe.cpp)
+else(WIN32)
+	list(APPEND LIVOX_SOURCES sdk_core/src/base/network/unix/network_util.cpp)
+	list(APPEND LIVOX_SOURCES sdk_core/src/base/wake_up/unix/wake_up_pipe.cpp)
+endif (WIN32)
+
+# Includes.
+set(LIVOX_PUBLIC_INCLUDES    
+    sdk_core/include
+    sdk_core/include/third_party/FastCRC
+    sdk_core/include/third_party/spdlog
+    sdk_core/include/third_party/cmdline
+)
+set(LIVOX_PRIVATE_INCLUDES sdk_core/src)
+
+# Add the Livox SDK library.
+add_library(${PROJECT_NAME} ${LIVOX_SOURCES})
+target_link_libraries(${PROJECT_NAME} Threads::Threads)
+target_sources(${PROJECT_NAME} PRIVATE ${LIVOX_SOURCES})
+target_include_directories(${PROJECT_NAME}
+    PUBLIC ${LIVOX_PUBLIC_INCLUDES} 
+    PRIVATE ${LIVOX_PRIVATE_INCLUDES})
+set_target_properties(${SDK_LIBRARY} 
+    PROPERTIES PUBLIC_HEADER "sdk_core/include/livox_def.h;sdk_core/include/livox_sdk.h")
+
+# Include headers.
+include_directories(sdk_core/include)
+
+# Install the catkin package.
+install(TARGETS
+	${PROJECT_NAME}
+    ARCHIVE DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
+    LIBRARY DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
+    RUNTIME DESTINATION ${CATKIN_PACKAGE_BIN_DESTINATION})
 
-message(STATUS "main project dir: " ${PROJECT_SOURCE_DIR})
-
-if (CMAKE_CROSSCOMPILING)
-	set(THREADS_PTHREAD_ARG
-		"PLEASE_FILL_OUT-FAILED_TO_RUN"
-		CACHE STRING "Result from TRY_RUN" FORCE)
-endif()
-
-if (UNIX)
-	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -pthread")
-endif(UNIX)
-
-add_subdirectory(sdk_core sdk_core)
-add_subdirectory(sample/hub)
-add_subdirectory(sample/lidar)
-add_subdirectory(sample/hub_lvx_file)
-add_subdirectory(sample/lidar_lvx_file)
-add_subdirectory(sample_cc/hub)
-add_subdirectory(sample_cc/lidar)
-add_subdirectory(sample_cc/trouble_shooting)
-add_subdirectory(sample_cc/lidar_utc_sync)
+# ## Install headers
+# install(FILES 
+# 	sdk_core/include/livox_def.h;
+# 	sdk_core/include/livox_sdk.h
+# 	DESTINATION ${CATKIN_GLOBAL_INCLUDE_DESTINATION}/${PROJECT_NAME})
```

Comment:
We're going to install the package to `/opt/ros` so we should install the headers.